### PR TITLE
FEAT: Update README to add DPO + CLIs

### PR DESCRIPTION
**Pull request by younesbelkada**
as per title, cc @lvwerra

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,13 +173,50 @@ reward = [torch.tensor(1.0)]
...
 ```
 
+### `DPOTrainer`
+
+`DPOTrainer` is a trainer that uses [Direct Preference Optimization algorithm](https://arxiv.org/abs/2305.18290). This is a basic example on how to use the `DPOTrainer` from the library. The `DPOTrainer` is a wrapper around the `transformers` Trainer to easily fine-tune reward models or adapters on a custom preference dataset.
+
+```python
+# imports
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from trl import RewardTrainer
```
Comment (younesbelkada) [inline README.md:183]:
```suggestion
from trl import DPOTrainer
```

**Review by lvwerra — APPROVED** [review]
